src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPDocumentManager.cs

```diff
@@ -92,10 +93,15 @@ public override void UntrackDocument(ITextBuffer buffer)
                 return;
             }
 
-            _documents.Remove(uri);
-
-            var args = new LSPDocumentChangeEventArgs(lspDocument.CurrentSnapshot, @new: null, LSPDocumentChangeKind.Removed);
-            Changed?.Invoke(this, args);
+            if (_documents.TryRemove(uri, out _))
+            {
+                var args = new LSPDocumentChangeEventArgs(lspDocument.CurrentSnapshot, @new: null, LSPDocumentChangeKind.Removed);
+                Changed?.Invoke(this, args);
+            }
+            else
+            {
+                throw new InvalidOperationException($"Couldn't remove {uri.AbsolutePath}. This should never ever happen.");
```

Comment:
Instead of an exception this should be a `Debug.Fail` since nothing catastrophic can happen from us hitting this endpoint